lib/date_time/date_time_ntp.c

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <net/sntp.h>
+#include <net/socketutils.h>
+
+#if defined(CONFIG_LTE_LINK_CONTROL)
+#include <modem/lte_lc.h>
+#endif
+
+#include <logging/log.h>
+
+LOG_MODULE_DECLARE(date_time, CONFIG_DATE_TIME_LOG_LEVEL);
+
+#define UIO_IP      "ntp.uio.no"
+#define GOOGLE_IP_1 "time1.google.com"
+#define GOOGLE_IP_2 "time2.google.com"
+#define GOOGLE_IP_3 "time3.google.com"
+#define GOOGLE_IP_4 "time4.google.com"
+
+#define NTP_DEFAULT_PORT "123"
+
+struct ntp_servers {
+	const char *server_str;
+	struct sockaddr addr;
+	socklen_t addrlen;
+};
+
+struct ntp_servers servers[] = {
+	{.server_str = UIO_IP},
+	{.server_str = GOOGLE_IP_1},
+	{.server_str = GOOGLE_IP_2},
+	{.server_str = GOOGLE_IP_3},
+	{.server_str = GOOGLE_IP_4}
+};
+
+static struct sntp_time sntp_time;
+
+static int sntp_time_request(struct ntp_servers *server, uint32_t timeout,
+			     struct sntp_time *time)
+{
+	int err;
+	static struct addrinfo hints;
+	struct addrinfo *addrinfo;
+	struct sntp_ctx sntp_ctx;
+
+	if (IS_ENABLED(CONFIG_DATE_TIME_IPV6)) {
+		hints.ai_family = AF_INET6;
+	} else {
+		hints.ai_family = AF_INET;
+	}
+	hints.ai_socktype = SOCK_DGRAM;
+	hints.ai_protocol = 0;
```

Comment:
This is only used if `server->addrlen == 0`, so this can be moved inside the `if`. Unless the code is modified so that `getaddrinfo()` is called every time, as I suggested in another comment.

Reply:
I did this now that I'm not sure yet if bigger refactoring of DNS queries will be done